evalml/data_checks/highly_null_rows_data_check.py

```diff
@@ -0,0 +1,85 @@
+from evalml.data_checks import (
+    DataCheck,
+    DataCheckAction,
+    DataCheckActionCode,
+    DataCheckMessageCode,
+    DataCheckWarning
+)
+from evalml.utils import _convert_woodwork_types_wrapper, infer_feature_types
+
+
+class HighlyNullRowsDataCheck(DataCheck):
```

Comment:
I wonder if we can share some code with `HighlyNullDataCheck`. The only computational difference I see is

`(X.isnull().mean()).to_dict()` vs `(X.isnull().mean(axis=1)).to_dict()`.

I'd be happy to punt on this though, doesn't seem super pressing.

Reply:
I thought about this as well but I felt that it fits better into `DefaultDataChecks` and `DataChecks` better as separate data checks. I don't feel too strongly on this though - happy to see what others think!